### ISSUE TITLE
Some additional discussion about training data rendering.

### ISSUE THREAD

**Issue by bbbbby-99** [SiTH-Diffusion/SiTH#25]
I attempted to use build_dataset.py to further render multi-view depth maps and modified the **rasterize** function:

```
def rasterize(camera, V, F, args):
    vertices_camera = camera.extrinsics.transform(V)
    face_vertices_camera = kal.ops.mesh.index_vertices_by_faces(
                            vertices_camera, F)
    face_normals_z = kal.ops.mesh.face_normals(face_vertices_camera,unit=True)[..., -1:].contiguous()
    proj = camera.projection_matrix()[0:1]
    homogeneous_vecs = kal.render.camera.up_to_homogeneous(
        vertices_camera
    )[..., None]
    vertices_clip = (proj @ homogeneous_vecs).squeeze(-1)
    rast = nvdiffrast.torch.rasterize(
        glctx, vertices_clip, F.int(),
        (args.size, args.size), grad_db=False
    )
    rast0 = torch.flip(rast[0], dims=(1,))
    hard_mask = rast0[:, :, :, -1:] != 0
# Make sure tensors are contiguous before interpolation
vertices_z = vertices_camera[:, 2:3].contiguous()
rast_contiguous = rast0.contiguous()
F_contiguous = F.int().contiguous()

# Interpolate depth
depth = nvdiffrast.torch.interpolate(
    vertices_z, rast_contiguous,
    F_contiguous
)[0]
depth_map = torch.flip(depth, dims=(0,)).contiguous()

return rast0, hard_mask, depth_map
```

To validate the rendering results, I normalized and visualized the depth maps, but they appear to be incorrect, as shown below:

```
  # Save depth data
  depth = depth_map[i, ..., 0]
  valid_mask = mask[i].squeeze()

  # Save raw depth values
  depth_data = depth.clone()
  depth_data[~valid_mask] = 0
  np.save(os.path.join(dirs['depth_raw'], f'depth_raw_{i:04d}.npy'), depth_data.cpu().numpy())

  # Save depth visualization
  img = torch.zeros((args.size, args.size, 4))
  depth_min = depth[mask[i].squeeze()].min()
  depth_max = depth[mask[i].squeeze()].max()
  depth_normalized = (depth - depth_min) / (depth_max - depth_min + 1e-8)

  img[..., 0] = depth_normalized
  img[..., 1] = depth_normalized
  img[..., 2] = depth_normalized
  img[..., 3] = valid_mask

  depth_img = Image.fromarray((255 * img).numpy().astype(np.uint8), mode='RGBA')
  depth_img.save(os.path.join(dirs['depth'], f'depth_{i:04d}.png'))
```
![depth_0021](https://github.com/user-attachments/assets/bb85799d-430b-4b72-9d00-fa797e98456f)
![depth_0000](https://github.com/user-attachments/assets/3ac028a4-ac0c-4daf-ac75-0aff396eab15)

I understand this might be additional work, but I would appreciate if you could help me understand why this issue occurred. Thank you.
